### PR TITLE
JAMES-3720 Fix temporary file leak in email reprocessing

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/MailRepositoryStoreService.java [hunk 1]
```diff
@@ -86,13 +87,21 @@ public MailRepository createMailRepository(MailRepositoryPath repositoryPath, St
     }
 
     public Optional<MailDto> retrieveMail(MailRepositoryPath path, MailKey mailKey, Set<AdditionalField> additionalAttributes) throws MailRepositoryStore.MailRepositoryStoreException, MessagingException, InaccessibleFieldException {
-        return fecthMail(path, mailKey)
-            .map(Throwing.function((Mail mail) -> MailDto.fromMail(mail, additionalAttributes)).sneakyThrow());
+        Optional<Mail> mail = fecthMail(path, mailKey);
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/MailRepositoryStoreService.java:90]: Can we fix while we are at it the typo of that method? `fecthMail` -> `fetchMail`? :)

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
read it

**Review by Arsnael — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by ottoka** [comment]
I wonder about RabbitMQ requeue, will this affect the process described in remote-delivery-error-handling.adoc for distributed servers? If so, will this also make it to the 3.7.x branch for an eventual 3.7.1 release?

**Comment by chibenwa** [comment]
> I wonder about RabbitMQ requeue, will this affect the process described in remote-delivery-error-handling.adoc for distributed servers? 

No in term of procedures it's the same.

The only noticeable difference is that you will not end up with tones of never deleted temporary files if you ever run it.

> If so, will this also make it to the 3.7.x branch for an eventual 3.7.1 release?

Contribution welcomed ;-)

Also we are putting efforts on a leak detector for having a systematic way of dealing with the issue, as well as a safety net for mailet extensions writer cf https://github.com/apache/james-project/pull/911 ...

**Comment by ottoka** [comment]
Tonnes of never deleted temporary files is what I was worrying about :(

**Comment by chibenwa** [comment]
> Tonnes of never deleted temporary files is what I was worrying about :(

Yes for sure, this changeset have a highly beneficial impact to queue/mailrepository management operations, and fixed a long standing issue...

**Comment by ottoka** [comment]
Well, for the remote delivery error use case it runs in a cron-job every hour, so not particularly rarely done. Seems like I will be doing a backport soon :)

Re System property, I see this only used in MimeMessageInputStreamSource/MimeMessageWrapper. But there are several places which just do new FileBackedOutputStream(SOME_CONSTANT). I wonder, should they use this property as well?

**Comment by chibenwa** [comment]
> Well, for the remote delivery error use case it runs in a cron-job every hour

the metric here is the count of eemail above 100k. if you have few emails failing remote delivery you will have few problems...

**Pull request merged**